pipelines.go

```diff
@@ -55,6 +56,60 @@ func (p *Pipeline) Lock() *Pipeline {
 	return p
 }
 
+func (p *Pipeline) ValidateRefIds() error {
+
+	var refidsSet = make(map[string]bool)
+	var requisitestagerefSet = make(map[string]bool)
+
+	//TODO: What should I return if stages do not exists? I think is valid to not have
+	//Check if there are stages, if not then do not process anything
+	if p.Stages != nil && len(p.Stages) > 0 {
+		//Iterate all stages
+		for _, stageMap := range p.Stages {
+
+			//ALL stages should have a redId
+			if _, exists := stageMap["refId"]; !exists {
+				//TODO: I think this should fail since refId is mandarory?
+				return errors.New("refId is a mandatory field for stages")
+			}
+
+			refId := stageMap["refId"].(string)
+
+			if _,exists := refidsSet[refId]; exists {
+				//TODO: Change this error message
```

Comment:
Change the error message to what?

Reply:
Not sure, I will ask Noah for help with the error messages

Reply:
Changed as per @noahi